components/observations/ObservationImageEditView.tsx

```diff
@@ -0,0 +1,122 @@
+import {zodResolver} from '@hookform/resolvers/zod';
+import {useHeaderHeight} from '@react-navigation/elements';
+import {Button} from 'components/content/Button';
+import {NetworkImage} from 'components/content/carousel/NetworkImage';
+import {VStack} from 'components/core';
+import {TextField} from 'components/form/TextField';
+import {BodySemibold, BodySm, Title3Semibold} from 'components/text';
+import React, {useCallback} from 'react';
+import {FormProvider, useForm} from 'react-hook-form';
+import {KeyboardAvoidingView, Platform, ScrollView, StyleSheet} from 'react-native';
+import {SafeAreaView, useSafeAreaInsets} from 'react-native-safe-area-context';
+import {z} from 'zod';
+
+interface CaptionFormData {
+  caption: string;
+}
+
+interface Props {
+  uri: string;
+  onSetCaption: (caption: string) => void;
+  initialCaption?: string;
+}
+
+const DefaultCaptionData: CaptionFormData = {
+  caption: '',
+};
+
+const captionForm = z.object({
+  caption: z.string(),
+});
+
+const BEHAVIOR = Platform.OS === 'ios' ? 'padding' : 'height';
+const EDGE_MODE = {top: 'off', left: 'additive', right: 'additive', bottom: 'additive'} as const;
+
+const IMAGE_WIDTH = (4 * 140) / 3;
+const IMAGE_HEIGHT = 140;
+
+export const ObservationImageEditView: React.FC<Props> = ({uri, onSetCaption, initialCaption}) => {
+  const formContext = useForm<CaptionFormData>({
+    defaultValues: initialCaption ? {caption: initialCaption} : DefaultCaptionData,
+    resolver: zodResolver(captionForm),
+    mode: 'onBlur',
+    shouldFocusError: false,
+    shouldUnregister: true,
+  });
+
+  const onSubmitPress = useCallback(() => {
+    formContext
+      .trigger()
+      .then(() => {
+        formContext
+          .handleSubmit(values => {
+            onSetCaption(values.caption);
+          })()
+          .catch(e => {
+            throw e;
+          });
+      })
+      .catch(e => {
+        throw e;
+      });
+  }, [formContext, onSetCaption]);
```

Comment:
I broke the pattern and didn't use `async`/`await` here because technically using `async` for this function is "wrong". The way it's used in this project with callback props results in unhandled rejected promises ([an example](https://github.com/NWACus/avy/blob/d3bcba00a7ead135207ee044b7def919cad1b158/components/screens/menu/AboutScreen.tsx#L30-L32)].

Another pattern I saw was using an immediately evaluated async function.

The `.then` and `.catch` methods of a `Promise` are pretty straight forward but maybe not preferred in this project?

I suppose it's done this because Sentry should be registering an event listener for unhandled rejections and they'll end up being captured there.

I guess I'm just sad that the "no un-awaited promises" rule makes one right code like this:

```ts
voyd (async () => {
  await myPromise;
})()
```